### PR TITLE
[DEVOPS-904] add release notes to the version.json

### PR DESCRIPTION
**Pull request by cleverca22**

### CODE REVIEW

File: iohk/UpdateProposal.hs [hunk 1]
```diff
@@ -113,14 +115,19 @@ parseUpdateProposalCommand = subparser $
...
+    revisionP = option (eitherReader (gitRevision . T.pack)) (long "revision" <> short 'r' <> metavar "SHA1" <> help "Daedalus revision to fetch")
+
+    blockVersionP :: Parser Text
+    blockVersionP = fmap T.pack $ strOption (long "block-version" <> short 'B' <> metavar "VERSION" <> help "Last known block version. Check the wiki for more info.")
+
+    voterIndexP :: Parser Int
+    voterIndexP = option auto (long "voter-index" <> short 'V' <> metavar "INTEGER" <> help "A number representing you, the vote proposer. Check the wiki for more info.")
+
+    releaseNotesP :: Parser (Maybe Text)
+    releaseNotesP = (Just . T.pack <$> strOption ( long "release-notes" <> metavar "RELEASE_NOTES" <> (help "Path to release notes (html)") <> (completer (bashCompleter "file") )))
```
Comment (rvl) [inline iohk/UpdateProposal.hs]: Better use the `FilePath` for all paths and `FP.fromText <$> option auto ...` to parse the option.

File: iohk/UpdateProposal.hs [hunk 2]
```diff
@@ -414,6 +426,7 @@ versionFile opts = cmdWorkPath opts </> "daedalus-latest-version.json"
 -- | Step 1. Init a new work directory
 updateProposalInit :: FilePath -> UpdateID -> UpdateProposalConfig1 -> Shell ()
 updateProposalInit top uid@(UpdateID _ date) cfg = do
+  liftIO $ print cfg
```
Comment (rvl) [inline iohk/UpdateProposal.hs]: don't need this debugging statement

File: iohk/UpdateProposal.hs [hunk 3]
```diff
@@ -423,7 +436,12 @@ updateProposalInit top uid@(UpdateID _ date) cfg = do
     True -> die "Config file already exists, stopping."
     False -> pure ()
   printf ("*** Creating blank template for date "%s%"\n") date
-  liftIO $ encodeFile (encodeString yaml) cfg
+  cfg' <- case (cfgReleaseNotes cfg) of
```
Comment (rvl) [inline iohk/UpdateProposal.hs]: could this be separated into a new function?

File: iohk/UpdateProposal.hs (hunk 1 above)
Comment (cleverca22) [reply iohk/UpdateProposal.hs]: the problem, is that i need the parser to return a `FilePath`, but then i want to store the `Text` within that file in the data record
and then i need 2 data records, for the parser, and the result after applying IO

Comment (rvl) [reply iohk/UpdateProposal.hs]: OK fair enough

**Comment by rvl** [comment]
It would be better if the code to read release notes and update the config were in a separate function.

**Comment by rvl** [comment]
I'm merging this now so that we can try it for the staging update proposal but the [docs](https://github.com/input-output-hk/iohk-ops/blob/master/docs/Daedalus-Installer-Update-Procedure.md) also need to be updated.

@cleverca22 Please update the docs in a separate PR

bors r+

**Pull request merged**